### PR TITLE
Update Threat Modeling Cheat Sheet

### PR DESCRIPTION
**Pull request by ben-dale**
This PR covers issue #300

Adds content to draft initially added by @ThunderSon, based on content from https://2018.open-security-summit.org/outcomes/tracks/threat-model/working-sessions/tm-cheatsheets/

### CODE REVIEW

**Comment by rbsec** [comment]
Hi @ben-dale,

I've had a quick read through this - I can see it's heavily based on the site you linked to.

While the do/don't model is good for people that already have some idea about the subject, it's harder for newcomers to understand. If someone's coming in who knows nothing about threat modelling (i.e, 99.9% of developers) then I think that the key questions that we need to try and answer are:

* What is threat modelling?
* Why should I care about it?
* Who needs to do it?
* What do I need to do?
* What is the end result/output?

I think that this is a difficult CS to pitch - the previous one was far too formal and not really approachable to anyone who didn't already know the topic.


I think my ideal level here would be a cheat sheet that you could give to a developer with some security knowledge (but no real previous experience doing threat modelling), that would allow them to achieve something useful (i.e, a tangible improvement in their security) within a few days.

Thoughts @ThunderSon @mackowski ?

**Comment by ThunderSon** [comment]
I agree to the above.
The information inside is truly the outcomes of the security summit that multiple threat modelers worked on to make it as direct as possible, to people that are wanting to tackle threat modeling, yet don't provide enough knowledge on the whole process.
Yes, I know that people need to just do threat modeling and not overthink things, but that isn't enough for those that are outside of that mentality.

Adding on what was discussed, I would advise to add the various methodologies that could be used (STRIDE, LINDUNN, etc.), and which fits where best.

@swierckx your input would be valuable.

**Comment by hazanasec** [comment]
I've found the friendliest "What is TM" to be: _Identify the likely threats to a system to inform the design of security countermeasures._ 

To get there I would also personally like to see a nod towards starting by having a solid, up to date architecture diagram as being fundamental (personally would like to C4 modeling mentioned), then to start it's as simple as just discussing and plotting some potential threats on top of the diagram, such as the STRIDE elements. From there you know begin to know your risks and can discuss and document your countermeasures.

I think stating the need to scope, DFD, writing test cases etc. comes much later with maturity and not something developers just starting to look at what TM is need to be hit with.

**Comment by ThunderSon** [comment]
I'd like to see a part related to this research:
https://insights.sei.cmu.edu/sei_blog/2018/12/threat-modeling-12-available-methods.html

Methods that could help the user to pick best what fits their needs. Some are old, and a bit no longer relevant. Maybe even discussing the overall things to look for per method, and then link to this. This is not the focus of the CS. The least that should be mentioned would be STRIDE and LINDUNN. We can see the rest and how they fit in as well.

**Comment by ben-dale** [comment]
@rbsec @ThunderSon @hazanasec thank you all for the input :+1: I have restructured the CS, taking it back to basics a bit, thinking more along the lines of "If I had to explain this to a team of people who had never done Threat Modeling before, how would I lay it out for them?". I'll continue to flesh out the sections over the coming days.

**Comment by rbsec** [comment]
This looks like a good skeleton - I'm looking forward to seeing how it develops.

File: cheatsheets_draft/Threat_Modeling_Cheat_Sheet.md [hunk 1]
```diff
@@ -23,7 +23,18 @@ Before anything else, you must agree with the team on what should be threat mode
...
+
+System diagrams drawn as part of a threat model should be kept fairly high level, concerned more with components and context and less concerned with exact implementations. It is very easy to become bogged down in the smaller details. You may also have less experienced members of the team present, so drawing high level diagrams will make it easier for them to contribute.
+
+[C4](https://c4model.com/) is a model that defines a set of patterns for documenting system architectures in a clear and consistent way. The different levels of C4 diagrams [can be seen here](https://c4model.com/#CoreDiagrams). Generally, Level 2 diagrams provide enough detail for threat modelling.
+
+TODO example C4
+TODO include DFD's, or is there enough with C4 alone?
```
Comment (mackowski) [inline cheatsheets_draft/Threat_Modeling_Cheat_Sheet.md:37]: I would like to add a line about DFDs as a lot of people use them and they are sometimes simpler to create ad hoc for threat modeling exercise

Comment (jmanico) [reply cheatsheets_draft/Threat_Modeling_Cheat_Sheet.md]: Just a bump here, I am reviewing PR's and assume this is still a work in progress?

Comment (ben-dale) [reply cheatsheets_draft/Threat_Modeling_Cheat_Sheet.md]: @jmanico yes! Sorry, I hope to get back to this soon and finish off 👍

**Comment by jthomerson** [comment]
@jmanico @ben-dale Sorry for what may be a dumb question that's answered somewhere that I'm not finding, so if it is, I apologize.

I'm reading through the [current threat modeling cheat sheet](https://cheatsheetseries.owasp.org/cheatsheets/Threat_Modeling_Cheat_Sheet.html) and wanted to submit a PR to fix some grammar issues and typos, and to make a few little improvements. But then I saw this PR, which appears to be rewriting that cheat sheet. I'm not real clear on how things in this draft folder relate to the published cheat sheets.

Should I go ahead and submit a PR to fix things on the existing cheat sheet? Or is that pointless if this one will soon replace it?

Thanks!

**Comment by jmanico** [comment]
Jeremy,

The edit to update the cheatsheet has been slow going. Would you mind 
still issuing a PR to fix existing typos? It may take a LONG time before 
this CS is really updated.

Does that work for you? I assume its not a huge effort?

Aloha, Jim

**Comment by mackowski** [comment]
@ben-dale do you still want to work on that?

**Comment by ben-dale** [comment]
@mackowski Unfortunately I've struggled to find the time with work and life, at the moment, to finish off this CS. Happy for it to be merged into drafts and handed over to someone else, otherwise I could probably pick it up in a couple of months. Sorry for the delays on this piece of work 👍

**Pull request merged**